pkg/cluster/generate.go

```diff
@@ -3,5 +3,5 @@ package cluster
 // Copyright (c) Microsoft Corporation.
 // Licensed under the Apache License 2.0.
 
-//go:generate go run ../../vendor/github.com/golang/mock/mockgen -destination=../util/mocks/$GOPACKAGE/$GOPACKAGE.go github.com/Azure/ARO-RP/pkg/$GOPACKAGE Interface
+//go:generate go run ../../vendor/github.com/golang/mock/mockgen --build_flags=--tags=aro -destination=../util/mocks/$GOPACKAGE/$GOPACKAGE.go github.com/Azure/ARO-RP/pkg/$GOPACKAGE Interface
```

Comment:
So maybe this init is in the wrong place?
Any particual reason we don't do this in `cmd/rp.go` (and `hacks/clusters`potentially)
```
if !platform.IsARO(){
    return fmt.Error("ARO-RP must be built, run, and tested with '-tags aro'")
}
```
Reason being:
1. Not everything needs and should be using this flag. Any hacks which don't need it should just run.
2. Panic is very drastic way to handle this, not sure if this is right method, especially in the init function. 

Don't know, just spitballing but panic does not looks right for sure.

Reply:
@m1kola wdyt? for me this does not look right

Reply:
> So maybe this init is in the wrong place?
> Any particual reason we don't do this in cmd/rp.go (and hacks/clusterspotentially)

I raised similar question here: https://github.com/Azure/ARO-RP/pull/1702/files#r696739641

But I did not explain the reason behind my suggestion. I think we need to put it to `cmd`, because there is a chance that we import something form the installer outside of `github.com/Azure/ARO-RP/pkg/cluster` package and if we do that and run the code without the build tag - `platform.IsARO()` will be returning `false` inside the intaller potentially causing incorrect behaviour.

Beucase `cmd` is the entry point to our codebase in prod (I'm not too woried about `./hack` - these are for dev only) - we can be failry confident that if we have this check in `cmd` - we run the correct build.

> and `hacks/clusters`  potentially

I think we are good here: I it doesn't import anything from the installer as far as I can see.

> Not everything needs and should be using this flag. Any hacks which don't need it should just run.

If we can avoid having to provide build tag to hack scripts - that would be great: will make our dev experience a bit easier. But I'm not too woried about it. The tag itself won't hurt if we compile everything with it.

> Panic is very drastic way to handle this, not sure if this is right method, especially in the init function.

I agree. I like the idea of doing `platform.IsARO()` check. If it is `false` - log error and exit with non-zero exit code.

Also using `platform.IsARO()` makes more sense, I think: we don't actually care if we have `aro` build tag. We care that `platform.IsARO()` returns `true` and this is what we should test for (in both unit test and in `init`/`main`/somewhere else). And this eliminates the need of having `// +build !aro`files.

Reply:
agreed :)

Reply:
This should now be as expected.